db/collection.go

```diff
@@ -16,16 +17,27 @@ type Collection struct {
 // NewCollection creates and returns a new collection with the given name and
 // model type. You should create exactly one collection for each model type. The
 // collection should typically be created once at the start of your application
-// and re-used.
-func (db *DB) NewCollection(name string, typ Model) *Collection {
-	return &Collection{
+// and re-used. NewCollection returns an error if a collection has already been
+// created with the given name for this db.
```

Comment:
Really I should have added this when we implemented our own version of transactions. The name of the collection determines which keys are part of it. If you have two collections with the same name and open a transaction on one of them, we can't guarantee consistency because the other collection still has access to the same keys.